sway-lib-std/src/alloc.sw

```diff
@@ -43,3 +43,25 @@ pub fn realloc<T>(ptr: raw_ptr, count: u64, new_count: u64) -> raw_ptr {
         ptr
     }
 }
+
+/// Allocates zeroed memory on the heap in individual bytes
+pub fn alloc_bytes(count: u64) -> raw_ptr {
+    asm(size: count, ptr) {
+        aloc size;
+        addi ptr hp i1;
+        ptr: raw_ptr
+    }
+}
+
+/// Reallocates the given area of memory in individual bytes
+pub fn realloc_bytes(ptr: raw_ptr, count: u64, new_count: u64) -> raw_ptr {
+    if new_count > count {
+        let new_ptr = alloc_bytes(new_count);
+        if count > 0 {
+            ptr.copy_bytes_to(new_ptr, count);
+        };
+        new_ptr
+    } else {
+        ptr
+    }
+}
```

Comment:
Yeah, this is annoying.  `__size_of<u8>` really should be 1.  Ugh.

Reply:
Yeah, I don't love this. It works but could be cleaner. 
I considered adding some conditional branching to the existing types, but there's no simple way to know if the function should treat `u8` as a single byte or as 8 bytes like everywhere else.